src/common/time/src/interval.rs

```diff
@@ -0,0 +1,708 @@
+            }
+            if self.minutes != 0 {
+                time_part.push_str(&format!("{}M", self.minutes));
+            }
+            if self.seconds != 0 {
+                time_part.push_str(&format!("{}S", self.seconds));
+            }
+            if self.microseconds != 0 {
+                let ms = self.microseconds.unsigned_abs();
+                time_part.push_str(&format!(".{:06}", ms));
+            }
+        } else {
+            time_part = "".to_owned();
+        }
+        if self.years != 0 {
+            result.push_str(&format!("{}Y", self.years));
+        }
+        if self.months != 0 {
+            result.push_str(&format!("{}M", self.months));
+        }
+        if self.days != 0 {
+            day.push_str(&format!("{}D", self.days));
+        }
+        result.push_str(&day);
+        result.push_str(&time_part);
+        result
+    }
+
+    /// Convert IntervalFormat to sql standard format string
+    pub fn to_sql_standard_string(self) -> String {
+        if self.is_zero() {
+            "0".to_owned()
+        } else if !self.has_time_part() && !self.has_day() {
+            get_year_month(self.months, self.years, true)
+        } else if !self.has_time_part() && !self.has_year_month() {
+            format!("{} 0:00:00", self.days)
+        } else if !self.has_year_month() && !self.has_day() {
+            get_time_part(
+                self.hours,
+                self.minutes,
+                self.seconds,
+                self.microseconds,
+                self.has_time_part_positive(),
+                true,
+            )
+        } else {
+            let year_month = get_year_month(self.months, self.years, false);
+            let time_interval = get_time_part(
+                self.hours,
+                self.minutes,
+                self.seconds,
+                self.microseconds,
+                self.has_time_part_positive(),
+                false,
+            );
+            format!("{} {:+} {}", year_month, self.days, time_interval)
+        }
+    }
+
+    /// Convert IntervalFormat to postgres format string
+    pub fn to_postgres_string(&self) -> String {
+        if self.is_zero() {
+            return "00:00:00".to_owned();
+        }
+        let mut result = "".to_owned();
+        let mut year_month = "".to_owned();
+        let mut day = "".to_owned();
+        let time_part = self.get_postgres_time_part();
+        if self.has_day() {
+            day = format!("{} days ", self.days)
+        }
+        if self.has_year_month() {
+            if self.years != 0 {
+                year_month.push_str(&format!("{} year ", self.years))
+            }
+            if self.months != 0 {
+                year_month.push_str(&format!("{} mons ", self.months));
+            }
+        }
+        result.push_str(&year_month);
+        result.push_str(&day);
+        result.push_str(&time_part);
+        result.trim().to_owned()
```

Comment:
Looks like we could simplify the following code
```suggestion
        let mut result = "".to_string();
        if self.has_year_month() {
            if self.years != 0 {
                result.push_str(&format!("{} year ", self.years));
            }
            if self.months != 0 {
                result.push_str(&format!("{} mons ", self.months));
            }
        }
        if self.has_day() {
            result.push_str(&format!("{} days ", self.days));
        }
        result.push_str(&self.get_postgres_time_part());
        result.trim().to_string()
```